docs/source/guides/logical_types_and_semantic_tags.ipynb

```diff
@@ -0,0 +1,629 @@
+    "\n",
+    "# Semantic Tags\n",
+    "\n",
+    "Here is the full set of Woodwork-defined semantic tags:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "supposed-cookie",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import woodwork as ww\n",
+    "ww.list_semantic_tags()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "extra-husband",
+   "metadata": {},
+   "source": [
+    "#### Standard Tags\n",
+    "Standard tags are associated with specific logical types. They are useful for indicating predefined categories that logical types might fall into.\n",
+    "- `'numeric'` - Is applied to any numeric logical type\n",
+    "    - **Uses**: Can select for just numeric columns when performing operations that require numeric operations\n",
+    "    - **Related Properties**: `series.ww.is_numeric`\n",
+    "- `'category'` - Is applied to any logical type that is categorical in nature\n",
+    "    - **Uses**: Can select for just categorical columns when performing operations that require categories\n",
+    "    - **Related Properties**: `series.ww.is_categorical`\n",
+    "\n",
+    "#### Index Tags\n",
+    "Index tags are added by Woodwork to a DataFrame when an `index` or `time_index` column is identified by the user. These tags have some special properties that are only confirmed to be true in the context of a DataFrame (so any Series with these tags may not have these properties).\n",
+    "- `'index'` - Indicates that a column is the DataFrame's index, or primary key\n",
+    "    - There will only be one index column\n",
+    "    - The contents of an index column will be unique\n",
+    "    - An index column will have any standard semantic tags associated with its logical type removed\n",
+    "    - The data in an index column will be reflected in the DataFrame's underlying index\n",
+    "- `'time_index'`\n",
+    "    - There will only be one time index column\n",
+    "    - A time index column will contain either datetime or numeric data\n",
+    "\n",
+    "#### Other Tags\n",
+    "This section lists any tags that Woodwork will neither add to a DataFrame nor take direct action upon, but that is recognized to have a special meaning.\n",
+    "\n",
+    "Additional tags beyond the ones Woodwork adds at initialization may be useful for a DataFrame's interpretability, so users are encouraged to add any tags that will allow them to use their data more efficiently. \n",
+    "- `'date_of_birth'` - Indicates that a datetime column should be parsed as a date of birth\n",
+    "\n",
+    "# Logical Types\n",
+    "\n",
+    "Below are all of the Logical Types that Woodwork defines. We will now walk through each of these types with some examples and suggested uses."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "sunset-chassis",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import woodwork as ww\n",
+    "ww.list_logical_types()"
```

Comment:
You've brought up a lot of good points in these comments about the functionality of `list_logical_types` and `list_semantic_tags` as they're used in our docs (we also use them in other places in the woodwork docs, and will likely be using them in featuretools docs). I think it'll be enough changes that it might be better to make these improvements in a separate issue.

Thoughts @gsheni @thehomebrewnerd

Reply:
I'm fine with addressing these comments in a separate issue and MR.

Reply:
made a separate issue: #1091